File: hackerrank/easy/0056-flatland-space-stations/python/solution.py
## solution.py

from typing import List

def flatlandSpaceStations(n: int, c: List[int]) -> int:
    station_count = len(c)
    if station_count == 0:
        # Constraints guarantee at least one station, but handle safely.
        return 0

    if station_count == n:
        return 0

    sorted_stations = sorted(c)

    # Edge distances
    max_dist = max(sorted_stations[0], (n - 1) - sorted_stations[-1])

    # Inner gaps
    for i in range(1, station_count):
        gap = sorted_stations[i] - sorted_stations[i - 1]
        dist_in_gap = gap // 2
        if dist_in_gap > max_dist:
            max_dist = dist_in_gap

    return max_dist


# --- demo run ---
if __name__ == "__main__":
    print(flatlandSpaceStations(5, [0, 4]))                 # Expected: 2
    print(flatlandSpaceStations(6, [0, 1, 2, 4, 3, 5]))     # Expected: 0

File: hackerrank/easy/0056-flatland-space-stations/js/solution.mjs
// solution.mjs

export function flatlandSpaceStations(n, c) {
  const stationCount = c.length;
  if (stationCount === 0) {
    // According to constraints this should not happen, but keep it safe.
    return 0;
  }

  if (stationCount === n) {
    return 0;
  }

  const sorted = [...c].sort((a, b) => a - b);

  // Edge distances
  let maxDist = Math.max(sorted[0], (n - 1) - sorted[stationCount - 1]);

  // Inner gaps
  for (let i = 1; i < stationCount; i++) {
    const gap = sorted[i] - sorted[i - 1];
    const distInGap = Math.floor(gap / 2);
    if (distInGap > maxDist) {
      maxDist = distInGap;
    }
  }

  return maxDist;
}

// --- demo run ---
if (import.meta.url === `file://${process.argv[1]}`) {
  console.log(flatlandSpaceStations(5, [0, 4]));                  // Expected: 2
  console.log(flatlandSpaceStations(6, [0, 1, 2, 4, 3, 5]));      // Expected: 0
}
